src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,56 @@
+package lotto.controller;
+
+import lotto.domain.*;
+import lotto.generator.LottoGenerator;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+import java.util.List;
+
+public class LottoController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final LottoGenerator lottoGenerator;
+
+    public LottoController(InputView inputView, OutputView outputView, LottoGenerator lottoGenerator) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.lottoGenerator = lottoGenerator;
+    }
+
+    public void run() {
+        PurchaseAmount purchaseAmount = askAmount();
+        List<Lotto> lottos = lottoGenerator.issueLottos(purchaseAmount.getCount());
+        outputView.printLottos(lottos);
+        WinningLotto winningLotto = askWinningLotto();
+
+        LottoResult lottoResult = LottoResult.of(lottos, winningLotto, purchaseAmount);
+        outputView.printResults(lottoResult);
+    }
+
+    private PurchaseAmount askAmount() {
+        outputView.promptAmount();
+        int amount = inputView.readWithRetry(inputView::readNumber, outputView::printError);
+        return new PurchaseAmount(amount);
+    }
+
+    private WinningLotto askWinningLotto() {
+        Lotto winningNumbers = askWinningNumbers();
+        LottoNumber bonusNumber = askBonusNumber();
+
+        return new WinningLotto(winningNumbers, bonusNumber);
+    }
```

Comment:
캡슐화까지 이렇게 고려하신게 인상깊네요!

객체 생성의 책임까지 WinningLotto까지 가져간다면 더 완벽한 캡슐화일 것 같아요!

```suggestion
    private WinningLotto askWinningLotto() {
        Lotto winningNumbers = askWinningNumbers();
        LottoNumber bonusNumber = askBonusNumber();

        return WinningLotto.create(winningNumbers, bonusNumber);
    }
```

Reply:
객체 생성을 제대로 고려하지 못한게 너무 아쉽네요! 규일님 덕분에 배워갑니다!

Reply:
제가 생각하기에 생성자라는 워딩에서 이미 해당 객체의 생성을 책임진다는 의미를 내포하므로 생성자를 사용해도 충분히 캡슐화가 된 것 같아요!